### ISSUE TITLE
improvedLinefit in EHE filter (L1 online and L2) (Trac #1858)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1850]
It seems that the EHE filter uses the various sub-modules of improvedLinefit separately (both in the online EHE filter and in L2). Since the change to merge all improvedLinefit modules into the linefit module and merging them into a single module these scripts in L2 are currently broken.

I would suggest to restore improvedLinefit for now so that these scripts are usable again (linefit does not need to be changed). This would be a short-term fix. In the slightly longer term we should just provide the original sub-modules in linefit.

This currently blocks a potential pass2 reprocessing.

Migrated from https://code.icecube.wisc.edu/ticket/1858
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:13:18", 
    "description": "It seems that the EHE filter uses the various sub-modules of improvedLinefit separately (both in the online EHE filter and in L2). Since the change to merge all improvedLinefit modules into the linefit module and merging them into a single module these scripts in L2 are currently broken.\n\nI would suggest to restore improvedLinefit for now so that these scripts are usable again (linefit does not need to be changed). This would be a short-term fix. In the slightly longer term we should just provide the original sub-modules in linefit.\n\nThis currently blocks a potential pass2 reprocessing.", 
    "reporter": "claudio.kopper", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067198646291", 
    "component": "combo reconstruction", 
    "summary": "improvedLinefit in EHE filter (L1 online and L2)", 
    "priority": "blocker", 
    "keywords": "", 
    "time": "2016-09-12T14:57:04", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20160912T16:35:38: gmaggi commented:

> As commented to Claudio. Having a look into  (for example) http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/projects/filterscripts/trunk/python/offlineL2/level2_HitCleaning_EHE.py#L284 , We could add some parameters in the configuration of the NewLineFit module, like: RunOnlyDelayCleaningEHE. Nevertheless, we would need to change a bit the filter scripts.
> 
>

**Comment by icecube-trac** [comment]
Trac update at 20160912T17:07:13: olivas commented:

> This shouldn't be an issue for the last release of IceRec (V05-00-02).  If there's something on the trunk that's needed for pass2 processing it might be easier to get that into an incremental release V05-00-03.

**Comment by icecube-trac** [comment]
Trac update at 20160916T18:28:36: 
* olivas changed status from "new" to "assigned"
* olivas changed component from "cmake" to "icerec"
* olivas changed owner from "olivas" to "gmaggi"

**Comment by icecube-trac** [comment]
Trac update at 20160916T18:49:00: 
* gmaggi commented:

> I will work on that and will present the modifications to Claudio and Alex before to commit, to check whether they agree. I will do my best to have it at the beginning of the next week. 


* gmaggi changed status from "assigned" to "accepted"

**Comment by icecube-trac** [comment]
Trac update at 20161003T15:15:49: 
* david.schultz commented:

> I talked with Olivas at the simulation workshop, and the "best" way forward is to not make any functional changes to the EHE scripts and go back to having separate modules and a tray segment for improved linefit.
> 
> Since we want this for pass2 within the next day or two, I'll make that happen within the next few hours.


* david.schultz changed owner from "gmaggi" to "david.schultz"

**Comment by icecube-trac** [comment]
Trac update at 20161003T15:40:39: gmaggi commented:

> As mentioned by email before the collaboration meeting, I made the modifications in my working copies of the new-linefit and filterscripts. After the meeting I have been working on testing the changes, and they do not affect the filter scripts performance. To test them I started with PFRawfiles: and the scripts are in:
> http://code.icecube.wisc.edu/projects/icecube/browser/IceCube/sandbox/gmaggi/icerec/patch/filterscripts
> Then, I produce some plots as well.
> 
> Of course, if it is better not to do any functional changes to the EHE scripts, we may go back to the previous improvedLineFit and LienFit. 
> 
>

**Comment by icecube-trac** [comment]
Trac update at 20161003T16:05:22: david.schultz commented:

> The problem with making functional changes to EHE is that they will take months to get approval, if ever.  We want pass2 to run within the next few days.

**Comment by icecube-trac** [comment]
Trac update at 20161004T16:14:30: 
* david.schultz changed status from "accepted" to "closed"
* david.schultz commented:

> In changeset 150514:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="150514"
> revert r148760-HEAD for linefit. instead, just copy the improvedLinefit modules to linefit and change all the names.  this quick and dirty approach will work for pass2 EHE.  in the long term we could consider asking EHE to change the way their scripts work and go back to a single module with cleaning functions. closes #1850
> ```


* david.schultz changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:13:18: 
* olivas commented:

> Milestone September 2016 deleted


* olivas changed milestone from "September 2016" to ""
